cli/lsp/documents.rs

```diff
@@ -1057,61 +1100,99 @@ impl DocumentsInner {
           .map(|im| JsxResolver::new(im, self.maybe_import_map.clone()))
       })
       .flatten();
-    if let Some(Ok(Some(imports))) =
-      maybe_config_file.map(|cf| cf.to_maybe_imports())
-    {
-      for (referrer, dependencies) in imports {
-        let dependencies =
-          dependencies.into_iter().map(|s| (s, None)).collect();
-        let module = SyntheticModule::new(
-          referrer.clone(),
-          dependencies,
-          self.get_maybe_resolver(),
-        );
-        self.imports.insert(referrer, module);
-      }
-    }
+    self.imports = Arc::new(
```

Comment:
This seems more correct, but maybe not? I believe we should replace `imports` when a configuration file changes rather than always inserting. Maybe I'm misunderstanding?

Reply:
I considered it, and there is no _harm_ in inserting, in that they are simply in memory and available, meaning that there could be old references or state referring to it, and therefore less likely to cause exotic errors.